File: src/main/java/imaxct/dao/impl/CourseDaoImpl.java
package imaxct.dao.impl;

import imaxct.dao.ICourseDao;
import imaxct.domain.Course;
import org.springframework.stereotype.Repository;

import java.util.List;

/**
 * Created by maxct on 2017/4/9.
 */
@Repository
public class CourseDaoImpl extends BaseDao<Course> implements ICourseDao {

    public boolean addCourse(Course course) {
        return this.create(course);
    }

    public boolean deleteCourse(Course course) {
        return this.delete(course);
    }

    public boolean updateCourse(Course course) {
        return this.update(course);
    }

    public List<Course> getAllCourses() {
        return this.listPage(0, 20, "from Course order by id desc");
    }

    public List<Course> getCourseFromId(int id){
        return this.listPage(0, 20, "from Course where id<? order by id desc", id);
    }
    public Course getCourseById(int id) {
        return this.find(Course.class, id);
    }
}

File: src/main/java/imaxct/dao/impl/CourseDaoImpl.kt
package imaxct.dao.impl

import imaxct.dao.ICourseDao
import imaxct.domain.Course
import org.springframework.stereotype.Repository

@Repository
class CourseDaoImpl : BaseDao<Course>(), ICourseDao {
    override fun allCourses(): List<Course> = this.list("from Course order by id desc")

    override fun addCourse(course: Course): Boolean = this.create(course)

    override fun deleteCourse(course: Course): Boolean = this.delete(course)

    override fun updateCourse(course: Course): Boolean = this.update(course)

    override fun getAllCourses(): List<Course> {
        return this.listPage(0, 30,
                "from Course where endDate > NOW() or endDate is null order by id desc")
    }

    override fun getCourseFromId(id: Int): List<Course> {
        return this.listPage(0, 30,
                "from Course where (endDate > NOW() or endDate is null) and id<? order by id desc", id)
    }

    override fun getCourseById(id: Int): Course = this.find(Course::class.java, id)
}
